models/issues/issue.go

```diff
@@ -2503,3 +2539,90 @@ func DeleteOrphanedIssues(ctx context.Context) error {
 func (issue *Issue) HasOriginalAuthor() bool {
 	return issue.OriginalAuthor != "" && issue.OriginalAuthorID != 0
 }
+
+var (
+	// Time estimate match regex
+	rTimeEstimateOnlyHours = regexp.MustCompile(`^([\d]+)$`)
+	rTimeEstimateWeeks     = regexp.MustCompile(`([\d]+)w`)
+	rTimeEstimateDays      = regexp.MustCompile(`([\d]+)d`)
+	rTimeEstimateHours     = regexp.MustCompile(`([\d]+)h`)
+	rTimeEstimateMinutes   = regexp.MustCompile(`([\d]+)m`)
+)
+
+// TimeEstimateFromStr returns time estimate in seconds from formatted string
+func (issue *Issue) TimeEstimateFromStr(timeStr string) int64 {
```

Comment:
These functions should be moved to some util package. Also don't we have a similar function when PR is merged/commit pushed that also parses time spent in format ex `@1h`?

Reply:
TBH, before asking for change, is there a conclusion that this feature could be merged or not?

If there is a consensus that this PR will be merged soon, then I can also help to review.

If there is no consensus, even if the requested changes are done, would a PR still get stale?

Reply:
At least it's in 1.20 milestone. So I have hope it'll be merged. I have no clue where to discuss it. And as I sad in the issue itself, current time tracking is meaningless for any corporate use if there's no estimation

Reply:
I'm not against adding this feature, I would probably even use it :)